droid-hal-device.inc

```diff
@@ -829,20 +785,62 @@ fi
 %endif
 %endif
 
+%pre users
+#Compatibility with old droid-hal version - convert previous format
+#of droid-specific user/group information to current one.
+cd %{_libdir}/droid
+if [ $1 -gt 1 -a \
+     -e droid-user-remove.sh.installed ]; then
+  echo '#!/bin/sh' > droid-user-remove.sh.installed
+fi
+
+%post users
+cd %{_libdir}/droid
+
+# NOTE: We do not delete groups/users in any case since that would cause issues
+# with users being removed from groups where they should really be in.
+
+echo creating droid users and groups
+[ -e droid.ids ] && cat droid.ids | while read name id; do
+    if ! getent group $name; then
+        echo "Group $name did not exist yet"
+        groupadd -g $id $name || :
+    else
+        echo "Group $name already existed"
+        groupmod -g $id $name || :
+    fi
+    if ! getent passwd $name; then
+        echo "User $name did not exist yet"
+        useradd -M -N -s /sbin/nologin -d / -u $id -g $id $name || :
+    else
+        echo "User $name already existed"
+        usermod -s /sbin/nologin -d / -u $id -g $id $name || :
+    fi
+done
+
+# Now ensure default user has access to various subsystems this HA provides
+# These are the default android ones:
+/usr/bin/groupadd-user audio || :
+/usr/bin/groupadd-user graphics || :
+/usr/bin/groupadd-user system || :
```

Comment:
I don't think that default user should be added system

Reply:
> I don't think that default user should be added system

That should be filed as a separate bug.

Reply:
Thanks